bdb/cursor.c

```diff
@@ -8720,19 +8720,95 @@ static void *db_count(void *varg)
     return NULL;
 }
 
-int gbl_parallel_count = 0;
-int bdb_direct_count(bdb_cursor_ifn_t *cur, int ixnum, int64_t *rcnt)
+static int find_recnum_first_last(DBC *dbcp, int *rnum, int last)
+{
+    DBT dbt_key, dbt_data;
+    int recnum;
+    int rc;
+    memset(&dbt_key, 0, sizeof(dbt_key));
+    memset(&dbt_data, 0, sizeof(dbt_data));
+    dbt_key.flags |= DB_DBT_MALLOC;
+    dbt_data.flags |= DB_DBT_MALLOC;
+    rc = dbcp->c_get(dbcp, &dbt_key, &dbt_data, last ? DB_LAST : DB_FIRST);
+    if (rc)
+        return rc;
+    rc = comdb2_sql_tick();
```

Comment:
Do I need to call comdb2_sql_tick after every time I call c_get? db_count() does but that might just be cuz it's in a while loop